File: C# Examples/09 Drawing Automation/AutoDrawing/SolidWorksMacro.cs
using System;
using System.IO;
using System.Diagnostics;
using System.Windows.Forms;

using SolidWorks.Interop.sldworks;
using SolidWorks.Interop.swconst;


namespace AutoDrawing
{
    public partial class SolidWorksMacro
    {
public void main()
{
    //browse for a folder to process
    FolderBrowserDialog fbd = new FolderBrowserDialog;
    DialogResult myRes;
    fbd.Description = "Select a folder...";
    myRes = fbd.ShowDialog();
    string MyPath;
    if (myRes == DialogResult.OK)
    {
        MyPath = fbd.SelectedPath;
    }
    else
    {
        //no folder selected
        return;
    }
    //set the directory for parts
    DirectoryInfo MyDir = new DirectoryInfo(MyPath);
    if (MyDir.Exists)
    {
        //enter a valid template path
        string templatePath = "C:\\...\\Drawing.drwdot";
        FileInfo[] AllParts;
        AllParts = (FileInfo[])MyDir.GetFiles("*.sldprt");
        foreach (FileInfo MyFile in AllParts)
        {
            //Write the file path to the Immediate Window
            Debug.Print(MyFile.FullName);
            //open the model
            ModelDoc2 swDoc = openModel(MyFile.FullName);
            //create the drawing
            DrawingFromModel(templatePath, MyFile.FullName);
        }
    }            
}

private ModelDoc2 openModel(string modelPath)
{
    int docType;
    int errors = 0;
    int warnings = 0;
    switch (Path.GetExtension(modelPath).ToUpper())
    {
        case ".SLDPRT":
            docType = (int)swDocumentTypes_e.swDocPART;
            break;
        case ".SLDASM":
            docType = (int)swDocumentTypes_e.swDocASSEMBLY;
            break;
        case ".SLDDRW":
            docType = (int)swDocumentTypes_e.swDocDRAWING;
            break;
        default:
            docType = (int)swDocumentTypes_e.swDocNONE;
            break;
    }
    ModelDoc2 swDoc = swApp.OpenDoc6(modelPath, docType,
        (int)swOpenDocOptions_e.swOpenDocOptions_Silent, 
        "", errors, warnings);
    return swDoc;
}
public void DrawingFromModel(string drawingTemplate, string modelPath)
{
    int longstatus = 0;
    bool boolstatus;

    // 
    // New Document
    ModelDoc2 swDoc = null;
    swDoc = ((ModelDoc2)(swApp.NewDocument(drawingTemplate,0,0,0)));
    DrawingDoc swDrawing = null;
    swDrawing = (DrawingDoc)swDoc;
    boolstatus = swDrawing.Create3rdAngleViews(modelPath);
    SolidWorks.Interop.sldworks.View myView = null;
    myView = swDrawing.CreateDrawViewFromModelView3(modelPath, 
        "*Isometric", 0.451104, 0.2954, 0);            
    Array vAnnotations = null;            
    vAnnotations = ((Array)(swDrawing.InsertModelAnnotations3(0, 
        32776, true, true, false, true)));
    // 
    // Zoom To Fit
    swDoc.ViewZoomtofit2();
    // 
    // Save As
    string drawingPath = Path.ChangeExtension(modelPath, ".SLDDRW");
    longstatus = swDoc.SaveAs3(drawingPath, 0, 0);
    swApp.CloseDoc(drawingPath);
    swApp.CloseDoc(modelPath);
    return;
}

        // The SldWorks swApp variable is pre-assigned for you.
        public SldWorks swApp;

    }
}

File: 09 Drawing Automation/AutoDrawing/SolidWorksMacro.vb
Imports SolidWorks.Interop.sldworks
Imports SolidWorks.Interop.swconst
Imports System.Runtime.InteropServices
Imports System


Partial Class SolidWorksMacro
    Public Sub main()

        Dim swDoc As ModelDoc2 = Nothing
        Dim swPart As PartDoc = Nothing
        Dim swDrawing As DrawingDoc = Nothing
        Dim swAssembly As AssemblyDoc = Nothing
        Dim boolstatus As Boolean = False
        Dim longstatus As Integer = 0
        Dim longwarnings As Integer = 0
        swDoc = swApp.ActiveDoc

        '
        'New Document
        Dim swSheetWidth As Double
        swSheetWidth = 0.5588
        Dim swSheetHeight As Double
        swSheetHeight = 0.43179999999999996
        swDoc = swApp.NewDocument("C:\ProgramData\SolidWorks\SOLIDWORKS 2022\templates\Drawing.drwdot", 12, swSheetWidth, swSheetHeight)
        swDrawing = swDoc
        Dim swSheet As Object = Nothing
        swSheet = swDrawing.GetCurrentSheet()
        swSheet.SetProperties2(12, 12, 1, 1, False, swSheetWidth, swSheetHeight, True)
        swSheet.SetTemplateName("C:\ProgramData\SolidWorks\SOLIDWORKS 2023\lang\english\sheetformat\c - landscape." &
                "slddrt")
        swSheet.ReloadTemplate(True)
        swDrawing = swDoc
        boolstatus = swDrawing.Create3rdAngleViews("C:\...\PlanetGear.SLDPRT") 'enter a real file path
        Dim myView As View = Nothing
        swDrawing = CType(swDoc, DrawingDoc)
        myView = swDrawing.CreateDrawViewFromModelView3("C:\...\PlanetGear.SLDPRT",
                                                        "*Isometric", 0.4340922240373396R, 0.3055803757292882R, 0)  'enter a real file path
        boolstatus = swDrawing.ActivateView("Drawing View4")
        Dim vAnnotations As Array = Nothing

        vAnnotations = swDrawing.InsertModelAnnotations3(0, 32776, True, True, False, True)
        '
        'Zoom To Fit
        swDoc.ViewZoomtofit2()

        '
        'Save As
        longstatus = swDoc.SaveAs3("C:\...\PlanetGear.SLDDRW", 0, 0)  'enter a real file path
        'latest saveas method
        'swDoc.Extension.SaveAs3("C:\...\PlanetGear.SLDDRW", swSaveAsVersion_e.swSaveAsCurrentVersion, swSaveAsOptions_e.swSaveAsOptions_Silent, Nothing, Nothing, 0, 0)
    End Sub
    ''' <summary>
    ''' The SldWorks swApp variable is pre-assigned for you.
    ''' </summary>
    Public swApp As SldWorks
End Class
